TaskHouseUnitTests/EmpolyerControllerUnitTests.cs

```diff
@@ -0,0 +1,153 @@
+using Xunit;
+using TaskHouseApi.Controllers;
+using TaskHouseApi.Repositories;
+using Microsoft.AspNetCore.Mvc;
+using TaskHouseApi.Model;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace TaskHouseUnitTests
+{
+    public class EmpolyerControllerUnitTests
+    {
+        EmployerrsController controller;
+        IEmployerRepository repo;
+
+        public EmpolyerControllerUnitTests()
+        {
+            repo = new  FakeEmppolyerRepository();
+            controller = new EmployerrsController(repo);
+        }
+        
+        /// Test Get all
+        [Fact]
+        public async void EmpolyerController_Get_ReturnsAllElementsInRepo_WhenGivenNoParameters()
+        {
+            //Arrange and act
+            IEnumerable<Employer> result = await controller.Get();
+
+            //Asserts
+            Assert.Equal(3, result.Count());
+
+        }
+
+        ///Test Get with valid Id as parameter
+        [Fact]
+        public async void EmpolyerController_Get_ReturnsObjectReponseWithCorrectEmpolyer_WhenGivenValidId() 
+        { 
+            //arrange
+            int empolyerId = 2;
+
+            //Act
+            var result = await controller.Get(empolyerId);
+            var resultAsObject = await controller.Get(empolyerId) as ObjectResult;
+            var resultObject = resultAsObject.Value as Employer;
+
+            //Assert
+            var assertResult = Assert.IsType<ObjectResult>(result);
+            Assert.Equal(empolyerId, resultObject.Id);
+
+        }
+
+        ///Test Get with invalid Id as parameter
+        [Fact]
+        public  async void EmpolyerController_Get_ReturnsNotFound_WhenGivenInvalidId()
+        { 
+            //arrange
+            int empolyerId = 5000;
+
+            //Act
+            var result = await controller.Get(empolyerId) as NotFoundResult;
+            
+            //Assert
+            Assert.Equal(404, result.StatusCode);
+
+        }
+
+        
+
+        ///Test post with null empolyer object
+        [Fact]
+         public async void EmpolyerController_Create_ReturnsBadRequest_WhenGivenNullEmpolyer()
+        { 
+            //Arrange
+            Employer employer = null;
+
+            //Act
+            var result = await controller.Create(employer);
+
+            //Assert
+            Assert.IsType<BadRequestObjectResult>(result);
+            //Major inconsistencies in whether is return BadRequestResult or BadRequestObjectResult
+
+        }
+
+      
+        
+        ///Test put with invalid Id and null empolyer object
+        [Fact]
+        public async void EmpolyerController_Update_ReturnsBadRequestResult_WhenIdIsInvalidAndEmpolyerIsNull()
+        { 
+            //Arrange
+            Employer employer = null;
+            int id = 2600;
+
+            //Act
+            var result = await controller.Update(id, employer);
+
+            //Assert
+            var assertResult = Assert.IsType<BadRequestResult>(result); 
```

Comment:
Same as earlier, just perform the assertion, no need to store the result in a variable 👍